[ADP-3487] Make deposit `WalletState` more strict

Anviking opened the pull request:
- Make deposit wallet state more strict with StrictData
- Bump cardano-wallet-agda for new increased strictness from
    - https://github.com/cardano-foundation/cardano-wallet-agda/pull/116
    - https://github.com/cardano-foundation/cardano-wallet-agda/pull/117

Depends on #4856 (or tested with it at least, but probably mergeable without it as well)

## Results

### This PR

~600-800 MB memory usage when syncing with preprod

The heap profile suggests we still have a memory leak, but this is not immediately observable in the os's activity monitor, so this is a lot better than before.

![cardano-wallet-strictness-pr](https://github.com/user-attachments/assets/df9ef5c6-c5f5-4d21-963a-9c97dbf8fc82)

### Previously ([anviking/ADP-3487/NodeToClient](https://github.com/cardano-foundation/cardano-wallet/tree/anviking/ADP-3487/NodeToClient) )

9 GB and growing acording to os's activity monitor:

![cardano-wallet](https://github.com/user-attachments/assets/0e77e794-a439-425f-a686-f77f83e0a741)

### Issue Number

ADP-3494

Anviking committed:
Make deposit wallet state more strict with StrictData

Anviking committed:
Bump cardano-wallet-agda for new increased strictness

paolino commented on lib/customer-deposit-wallet/src/Cardano/Wallet/Deposit/Pure/State/Type.hs:
```diff
@@ -1,3 +1,4 @@
+{-# LANGUAGE StrictData #-}
 module Cardano.Wallet.Deposit.Pure.State.Type
     ( -- * Types
       WalletState (..)
```
can you pls remove the superfluous bangs in the datatype defs below ?

paolino reviewed (approved):
:+1:

Anviking committed:
Remove `!` made superflous by StrictData

Anviking committed:
Add 500ms threadDelay in scenario test submitTx

This is an unprincipled fix for the following macOS specific unit test
failure: https://buildkite.com/cardano-foundation/cardano-wallet/builds/8283#019377d9-7963-4ff1-8c42-f270f8669a0c
